go/vt/vttablet/tabletserver/repltracker/writer.go

```diff
@@ -38,77 +39,108 @@ import (
 	topodatapb "vitess.io/vitess/go/vt/proto/topodata"
 )
 
+// We identify these heartbeat configuration types:
+// - No configuration: the heartbeat writer is generally disabled and does not produce heartbeats (but see below).
+// - On-demand: on-demand heartbeat interval was specified.
+// - Always: the heartbeat writer is always on and produces heartbeats at a regular interval.
+type HeartbeatConfigType int
+
+const (
+	HeartbeatConfigTypeNone HeartbeatConfigType = iota
+	HeartbeatConfigTypeOnDemand
+	HeartbeatConfigTypeAlways
+)
+
 const (
 	sqlUpsertHeartbeat = "INSERT INTO %s.heartbeat (ts, tabletUid, keyspaceShard) VALUES (%a, %a, %a) ON DUPLICATE KEY UPDATE ts=VALUES(ts), tabletUid=VALUES(tabletUid)"
+
+	testKeyspaceShard = "test:0"
```

Comment:
I think that this should go in `writer_test.go`.

Reply:
Moved.